src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -119,5 +135,45 @@ internal override bool ShouldIncludeAccessibilityModifier(SyntaxNode typeNode)
             var typeDeclaration = typeNode as TypeDeclarationSyntax;
             return typeDeclaration.Modifiers.Any(m => SyntaxFacts.IsAccessibilityModifier(m.Kind()));
         }
+
+        internal override Solution GetSolutionWithSameFileUpdated(
+            Solution solution,
+            INamedTypeSymbol extractedInterfaceSymbol,
+            IEnumerable<ISymbol> includedMembers,
+            Dictionary<ISymbol, SyntaxAnnotation> symbolToDeclarationAnnotationMap,
+            List<DocumentId> documentIds,
+            SyntaxAnnotation typeNodeAnnotation,
+            DocumentId documentIdWithTypeNode,
+            CancellationToken cancellationToken)
+        {
+            var document = solution.GetDocument(documentIdWithTypeNode);
+            var originalRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+            var typeDeclaration = originalRoot.GetAnnotatedNodes<TypeDeclarationSyntax>(typeNodeAnnotation).Single();
+
+            document = document.WithSyntaxRoot(originalRoot.TrackNodes(typeDeclaration));
```

Comment:
overwriting documents in a method is super confusing.  Because now things like 'originalRoot' don't correspond to the 'document' they were originally created from.